acceptance/tests/ticket_4622_filebucket_diff_test.rb

```diff
@@ -1,86 +1,72 @@
 test_name "ticket 4622 filebucket diff test."
 confine :except, :platform => 'windows'
+skip_test 'skip test, no non-Windows agents specified' if agents.empty?
 
-def create_testfile(filename, contents)
-  create_remote_file(agent, filename, "#{contents}")
-end
-
-def get_checksum_from_backup(filename, bucket_locale)
-  on agent, puppet("filebucket backup #{filename} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
+def get_checksum_from_backup_on(host, filename, bucket_locale)
+  on host, puppet("filebucket backup #{filename} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
   output = result.stdout.strip.split(": ")
   checksum = output.last
   return checksum
 end
 
-def validate_diff(item_one, item_two, bucket_locale)
-  on agent, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
+def validate_diff_on(host, item_one, item_two, bucket_locale)
+  on host, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
   assert_match("-foo", result.stdout.strip)
   assert_match('+bar', result.stdout.strip)
   assert_match('+baz', result.stdout.strip)
```

Comment:
crap, missed those in the previous PR.  oh well.  next time we modify that
test??

On Fri, Jul 31, 2015 at 11:52 AM, Michael Smith <notifications@github.com>
wrote:

> In acceptance/tests/ticket_4622_filebucket_diff_test.rb
> <https://github.com/puppetlabs/puppet/pull/4120#discussion_r36004772>:
>
> >    output = result.stdout.strip.split(": ")
> >    checksum = output.last
> >    return checksum
> >  end
> >
> > -def validate_diff(item_one, item_two, bucket_locale)
> > -  on agent, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
> > +def validate_diff_on(host, item_one, item_two, bucket_locale)
> > +  on host, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
> >    assert_match("-foo", result.stdout.strip)
> >    assert_match('+bar', result.stdout.strip)
> >    assert_match('+baz', result.stdout.strip)
>
> Yeah... that also seems messy.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/4120/files#r36004772>.
>



-- 
Eric Thompson
eric.thompson@puppetlabs.com <erict@puppetlabs.com>
Senior Software Quality Assurance Engineer